File: leet-code/2410. Maximum Matching of Players With Trainers/MaximumMatchingOfPlayersWithTrainers2410.java
package com.example;

import java.util.Arrays;

public class MaximumMatchingOfPlayersWithTrainers2410 {
    public int matchPlayersAndTrainers(int[] players, int[] trainers) {
        Arrays.sort(players);
        Arrays.sort(trainers);
        int numberOfMatching = 0;
        int playerIdx = 0;
        for (int trainer : trainers) {
            if (playerIdx == players.length) {
                break;
            }
            int player = players[playerIdx];
            if (player > trainer) {
                continue;
            }
            numberOfMatching += 1;
            playerIdx += 1;
        }
        return numberOfMatching;
    }
}

File: leet-code/2410. Maximum Matching of Players With Trainers/MaximumMatchingOfPlayersWithTrainers2410.kt
package com.example

class MaximumMatchingOfPlayersWithTrainers2410 {
    fun matchPlayersAndTrainers(players: IntArray, trainers: IntArray): Int {
        players.sort()
        trainers.sort()
        var match = 0
        var playerIdx = 0
        var trainerIdx = 0
        while (playerIdx < players.size && trainerIdx < trainers.size) {
            if (players[playerIdx] <= trainers[trainerIdx]) {
                match++
                playerIdx++
            }
            trainerIdx++
        }
        return match
    }
}
